static/js/containers/CommentVoteForm.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+import React from 'react'
+import { connect } from "react-redux"
+import type { Dispatch } from 'redux'
+
+import { actions } from '../actions'
+import type { Comment } from '../flow/discussionTypes'
+
+type CommentVoteFormProps = {
+  comment: Comment,
+  dispatch: Dispatch,
+}
+
+class CommentVoteForm extends React.Component {
+  props: CommentVoteFormProps
+
+  downvote = () => {
+    const { comment, dispatch } = this.props
+    dispatch(actions.comments.patch(comment.id, {downvoted: !comment.downvoted}))
+  }
+
+  upvote = () => {
+    const { comment, dispatch } = this.props
+    dispatch(actions.comments.patch(comment.id, {upvoted: !comment.upvoted}))
+  }
+
+  render() {
+    const { comment } = this.props
+
+    return <div className="votes-form">
+      <div className="score">{comment.score}</div>
+      <div className="votes">
+        <button
+          className={`vote upvote ${comment.upvoted ? 'upvoted' : ''}`}
+          onClick={() => this.upvote()}
+        >
+          ⇑
+        </button>
+        <button
+          className={`vote downvote ${comment.downvoted ? 'downvoted' : ''}`}
+          onClick={() => this.downvote()}
+        >
+          ⇓
+        </button>
+      </div>
+    </div>
+  }
+}
+
+export default connect()(CommentVoteForm)
```

Comment:
what's the reason for implementing this as a separate component, as opposed to implementing functions on `PostPage` which are passed down to the `CommentTree` component?